internal/statesync/reactor.go

```diff
@@ -839,5 +951,46 @@ func (r *Reactor) fetchLightBlock(height uint64) (*types.LightBlock, error) {
 		},
 		ValidatorSet: vals,
 	}, nil
+}
+
+func (r *Reactor) waitForEnoughPeers(ctx context.Context, numPeers int) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-time.After(200 * time.Millisecond):
+			if r.peers.Len() >= numPeers {
+				return
+			}
+		}
+	}
```

Comment:
Optional alternative suggestion:
```go
t := time.NewTicker(200*time.Millisecond)
defer t.Stop()
for {
  select {
  case <-ctx.Done():
    return
  case <-t.C:
    if r.peers.Len() >= numPeers {
      return
    }
  }
}
```

Basically the same thing, but only one goroutine for the timer, and it cleans up after itself.

Does this need to synchronize on calling Len with concurrent peer updates?